File: iOS/EvianBlog/api/writings/SeriesAPI.swift
//
//  SeriesAPI.swift
//  EvianBlog
//
//  Created by Evian张 on 2020/5/18.
//  Copyright © 2020 Evian张. All rights reserved.
//

import Alamofire

import Foundation
import Combine

extension BlogAPI {
	func getSeries() -> AnyPublisher<[Series], BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Series.getSeries)
	}
	
	func getArticlesCountOfSeries(name: String) -> AnyPublisher<UInt, BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Series.getArticlesCountOfSeries(name))
	}
	
	func getArticlesOfSeries(name: String, pageIndex: UInt, pageSize: UInt) -> AnyPublisher<[ArticleMeta], BlogAPIError> {
		self.fetch(url: Endpoint.Writings.Series.getArticlesOfSeries(name, pageIndex, pageSize))
	}
}

File: Android/app/src/main/java/top/evian_zhang/evianblog/api/writings/SeriesAPI.kt
package top.evian_zhang.evianblog.api.writings

import top.evian_zhang.evianblog.api.ArticleMeta
import top.evian_zhang.evianblog.api.Series
import top.evian_zhang.evianblog.api.BlogAPI
import top.evian_zhang.evianblog.api.Endpoint

suspend fun BlogAPI.getSeries(): List<Series> {
    return this.fetch(Endpoint.Writings.Series.getSeries)
}

suspend fun BlogAPI.getArticlesCountOfSeries(name: String): Int {
    return this.fetch(Endpoint.Writings.Series.getArticlesCountOfSeries(name))
}

suspend fun BlogAPI.getArticlesOfSeries(name: String, pageIndex: Int, pageSize: Int): List<ArticleMeta> {
    return this.fetch(Endpoint.Writings.Series.getArticlesOfSeries(name, pageIndex, pageSize))
}
